### ISSUE TITLE
AssemblyLoadContext: Loading of assemblies from 'runtimes' folder

### ISSUE THREAD

**Issue by shvez** [dotnet/coreclr#25516]
@vitek-karas  you asked me here(https://github.com/dotnet/coreclr/issues/13277) to create a new issue. here it is.

In short:
My plugin application has a dependency from System.Diagnostics.PerformanceCounter.dll. I publish it as `dotnet publish -f netcoreapp3.0`

The issue is that if I just build my loader and start it, it loads plugin application without any issues. It takes a version of the library from `runtimes` folder.
if I publish Loader application like 'dotnet publish -f netcoreapp3.0 -r win-x64' (it is important that I use RID 'win-x64', not 'win10-x64') it loads  System.Diagnostics.PerformanceCounter.dll assembly from the plugin application folder. And this version is generating an exception that platform is not supported.

I've checked the trace file for failure case and found out that it can not load from runtimes folder because RID win10-x64 is not compatible with platform 'win' where System.Diagnostics.PerformanceCounter.dll is situated.  `runtimes\win\lib\netcoreapp2.0\System.Diagnostics.PerformanceCounter.dll` to be precise.  I do not see such message for NotPublished version

What do I do wrong? What should I do to get it working correctly? I need a version of my application which may run on windows, on linux, and on mac. 

here is netcore info:
```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview6-012264
 Commit:    be3f0c1a03

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview6-012264\

Host (useful for support):
  Version: 3.0.0-preview6-27804-01
  Commit:  fdf81c6faf

.NET Core SDKs installed:
  2.1.604 [C:\Program Files\dotnet\sdk]
  2.1.700 [C:\Program Files\dotnet\sdk]
  2.1.800-preview-009696 [C:\Program Files\dotnet\sdk]
  2.2.300 [C:\Program Files\dotnet\sdk]
  3.0.100-preview6-012264 [C:\Program Files\dotnet\sdk]
```

here are links to trace files and to sample application:
Loader app sample: https://www.dropbox.com/s/lqragpd26vfccr2/Loader.7z?dl=0
it contains Loader - app, Application - the plugin, AppInterfaces - common interface lib. 
'_out' folder which contains 'dev' folder with build results and 'publish' folder with results of publishing.

trace_p.7z(https://www.dropbox.com/s/vfwbzgfoyqf4umb/trace_p.log.7z?dl=0) - zipped version of trace for the version from 'publish' folder
trace.7z(https://www.dropbox.com/s/613qu1n8wy5a3rv/trace.log.7z?dl=0) - zipped version of trace for the version from 'dev' folder

to start dev version you may use visual studio. to start published version you have to go to '_out/publish/Loader' folder and start Loader.exe

**Comment by shvez** [comment]
From the traces I've attached to the initial post, I found out that in case of the published version the rid fallback graph is empty. I think this could be a reason for `The targeted framework does not support the runtime 'win10-x64'` message

**Comment by vitek-karas** [comment]
This is a known problem with the component loading in self-contained apps. The master issue is here: https://github.com/dotnet/core-setup/issues/6961. In your case you're specifically hitting https://github.com/dotnet/core-setup/issues/6960.

Unfortunately I'm currently not aware of a good workaround. The only one I can think of is to add some code into your `AssemblyLoadContext.Load` and handle this case specifically. Alternatively you could modify the build of your host app and manually add the RID fallback graph into its `.deps.json` - then it should work. If you only have one or two cases like this I would probably go with the custom code in the load context - it's not clean, the host app has to know about which libraries the plugin loads - but it's probably the least amount of trouble.

**Comment by shvez** [comment]
thank you for response

are you going to fix those issues before netcore 3 release?

**Comment by vitek-karas** [comment]
I don't think it will happen - the one you're hitting we might be able to as it's just an SDK change, doesn't require fixes in the host or the runtime. The other will very likely not happen as it requires changes both in SDK and the host.

**Comment by shvez** [comment]
well, one more possible solution is the publishing of plugin with an explicit platform. Some what like -r win-x64.
Although it may work for us (with a tradeoff of creating multiple copies of the same code in binary form) it may in some cases create troubles for our customers, who also provides plugins for our plugins
